docs/opcodes_details.md

```diff
@@ -0,0 +1,340 @@
+- **evmone**: Data is in `state.return_data.size()`
+- **openethereum**: Data is in `self.return_data.len()`
+
+no comments here, I can't imagine sizing could depend on anything.
+
+## `0x3e,RETURNDATACOPY`
+
+`0x3e,RETURNDATACOPY,"3 + 3 * ceil(amount / 32)",,3,0,This opcode has similar semantics to CALLDATACOPY, but instead of copying `
+
+- **geth**: Data is in `interpreter.returnData`
+- **evmone**: Data is in `state.return_data`
+- **openethereum**: Data is in `self.return_data`
+
+see `CALLDATACOPY`, everything is same just the data is from the code not input
+
+## `0x41,COINBASE` and friends
+
+`0x42,TIMESTAMP`, `0x43,NUMBER`, `0x44,DIFFICULTY`, `0x45,GASLIMIT` all follow COINBASE's example, just picking different fields from the same structures.
+
+`0x41,COINBASE,2,base,0,1,Get the block’s beneficiary address.,`
+
+- **geth**: Data is in `interpreter.evm.Context.Coinbase.Bytes()`
+- **evmone**: Data is in `state.host.get_tx_context().block_coinbase`
+- **openethereum**: Data is in `ext.env_info().author.clone()`
+
+## `0x50,POP`
+
+`0x50,POP,2,base,1,0,Remove item from stack.,`
+
+### Parameters
+
+- size of the stack?
+
+## `0x51,MLOAD`
+
+`0x51,MLOAD,3,verylow,1,1,Load word from memory.,`
+
+no comments here
+
+## `0x52,MSTORE` and the 8bit friend
+
+`0x52,MSTORE,3,verylow,2,0,Save word to memory,`
+`0x53,MSTORE8,3,verylow,2,0,Save byte to memory.,`
+
+### Parameters
+
+- amount of memory allocated (depending on the store index requested)
+
+## `0x56,JUMP`
+
+`0x56,JUMP,8,mid,1,0,Alter the program counter,`
+
+- **geth** - just validates and sets PC, simple
+- **evmone** - the code block pre-processing might heavily impact cost (ask PB **TODO**)
+- **openethereum** - jumptable is initialized on first JUMP!
+
+### Parameters
+
+- first vs subsequent jumps, esp. for large jumptables. **TODO**: specify this better.
+- size of jumptable
+
+## `0x57,JUMPI`
+
+`0x57,JUMPI,10,high,2,0,Conditionally alter the program counter.,`
+
+JUMPI only differs by checking the condition on the stack.
+Similar comments apply
+
+### Parameters
+
+- same as for JUMP
+- whether the jump actually happens or not!
+
+## `0x58,PC`
+
+`0x58,PC,2,base,0,1,Get the value of the program counter prior to the increment corresponding to this instruction.,`
+
+no comments here
+
+## `0x59,MSIZE`
+
+`0x59,MSIZE,2,base,0,1,Get the size of active memory in bytes.,`
+
+no comments here, I can't imagine sizing the memory variable could depend on anything.
+
+## `0x5a,GAS`
+
+`0x5a,GAS,2,base,0,1,"Get the amount of available gas, including the corresponding reduction for the cost of this instruction.",`
+
+Implementation differ due to different approaches to gas tracking, but I don't see any hidden parameters to account for.
+
+## `0x5b,JUMPDEST`
+
+`0x5b,JUMPDEST,1,,0,0,Mark a valid destination for jumps,`
+
+- **geth** noop
+- **evmone** not a noop, flushes some gas calculations for the code block. Note - JUMPDEST is rewriten to a `evmone`-specific OPX_BEGINBLOCK
```

Comment:
This is true for evmone/Advanced where gas cost and stack checks are done per instruction basic blocks. However, newer evmone version also has Baseline interpreter. If you are interested in checking the new one too I can rebase your instrumentation.